components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java

```diff
@@ -0,0 +1,207 @@
+import org.wso2.carbon.identity.application.authentication.framework.AuthenticatorFlowStatus;
+import org.wso2.carbon.identity.application.authentication.framework.config.builder.FileBasedConfigurationBuilder;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authenticator.basicauth.internal.BasicAuthenticatorServiceComponent;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.*;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Unit test cases for the OpenID Connect Authenticator.
+ */
+@PrepareForTest({IdentityTenantUtil.class, BasicAuthenticatorServiceComponent.class, User
+        .class, MultitenantUtils.class, FrameworkUtils.class, FileBasedConfigurationBuilder.class,
+        IdentityUtil.class, UserCoreUtil.class})
+public class BasicAuthenticatorTestCase extends PowerMockTestCase {
+
+    private HttpServletRequest mockRequest;
+    private HttpServletResponse mockResponse;
+    private AuthenticationContext mockAuthnCtxt;
+    private RealmService mockRealmService;
+    private UserRealm mockRealm;
+    private UserStoreManager mockUserStoreManager;
+    private AbstractApplicationAuthenticator mockAbstractApplicationAuthenticator;
+    private FileBasedConfigurationBuilder mockFileBasedConfigurationBuilder;
+
+    @Mock
+    private User user;
+
+    private BasicAuthenticator basicAuthenticator;
+
+    @BeforeTest
+    public void setup() {
+
+        basicAuthenticator = new BasicAuthenticator();
+
+    }
+
+    @Test
+    public void canHandlePassTestCase() {
+
+        mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.USER_NAME)).thenReturn("admin");
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.PASSWORD)).thenReturn("admin");
+        Assert.assertTrue(basicAuthenticator.canHandle(mockRequest), "Invalid can handle response for the request.");
+    }
+
+    @DataProvider(name = "usernameAndPasswordProvider")
+    public Object[][] getUsernameAndPassword() {
+
+        return new String[][]{{"admin", null}, {null, "admin"}, {null, null}};
```

Comment:
better to split cases to new lines, and to inlcude a scenarion involving empty strings